CRM/Remoteevent/Registration.php

```diff
@@ -196,21 +196,21 @@ public static function cannotRegister($event_id, $contact_id = null, $event_data
         }
 
         // check if max_participants set and NO waitlist:
-        if (!empty($event_data['max_participants']) && empty($event_data['has_waitlist'])) {
-            $registered_count = self::getRegistrationCount($event_id);
-            if ($registered_count >= $event_data['max_participants']) {
-                if (empty($event_data['event_full_text'])) {
-                    return E::ts("Event is booked out");
-                } else {
-                    return $event_data['event_full_text'];
-                }
+        if (!empty($event_data['max_participants'])) {
```

Comment:
what's the reasoning here to drop the "no waitlist" constraint?

Reply:
This refers to what I wrote in https://github.com/systopia/de.systopia.remoteevent/pull/82#issuecomment-2202640859:
> **Logic of `cannotRegister()` reasons**
> 
> * comparison with `max_participants`
>   
>   * currently includes `Positive` registrations only
>   * should be in line with how Core calculates that, which, according to [`\CRM_Event_BAO_Participant::eventFull()`](https://github.com/civicrm/civicrm-core/blob/70c85f9575f19299b5114a7712e3210c5f856a73/CRM/Event/BAO/Participant.php#L337-L339), is:
>     ```
>     // consider event is full when.
>     // 1. (count(is_counted) >= event_size) or
>     // 2. (count(participants-with-status-on-waitlist) > 0)
>     ```
>     so the check for `has_waitlist` can be removed and all `is_counted` registrations be compared to `max_participants` - for events _without_ a waitlist, there will be no waitlisted participants, and for events _with_ a waitlist, they will be correctly counted. Also, `Pending` registrations should always be counted, if they are `is_counted`, as well.

But I don't see how this is valid anymore and I think the constraint should not be removed. I'm a bit confused, because when I did this change, it really made sense to me.

Reply:
Re-introduced the constraint for no waitlist.